recipes/thrift/all/conanfile.py

```diff
@@ -78,7 +78,9 @@ def requirements(self):
             self.requires("libevent/2.1.12")
 
     def build_requirements(self):
-        if tools.os_info.is_windows:
+        build_os = self.settings_build.os if hasattr(conanfile, 'settings_build') else self.settings.os
+            
+        if build_os == "Windows":
```

Comment:
Hard to understand, actually I don't undestand it 😄 . It's unfortunate to have to write something like this in CCI recipes because package id are computed on a Linux machine.

Reply:
I believe it's a CI issue, recipe is fine.

Reply:
So I understand that:
- `settings_build` is available only with 2 profiles.
- Currently, conan client doesn't consider 1 profile (same build & host context) like 2 profiles with same build & host profiles, and `settings_build` is not available in this case, but it may change in upcoming versions.

It means that we may just use `self.settings_build.os` in the near future, without testing its existence and fallback to `self.settings.os`?
It also means that all occurrences of `tools.os_info.is_windows` in CCI should be replaced by `self.settings_build.os == "Windows"` when 1 profile will behave like 2 profiles.

Reply:
Hi, @SpaceIm . This is not a CI issue, it is a _bug_ in the recipe (probably `tools.os_info` was released long time ago, without paying too much attention to cross-building scenarios). Conan commands and output (given fixed inputs) cannot depend on the running machine.

Some people will different `profile:build` and running machine. For example, they run Conan locally, move the generated files to some build-system and trigger the compilation there (maybe cygwin, some subsystem,...). It is important that `build_requires` are aligned with `profile:build`.

As you have seen, checking for `self.settings_build.os` and falling back to `self.settings.os` is a workaround while we need to support one profile approach (as long as Conan v1 is supported). With two-profiles approach (or Conan v2) the fallback is no longer needed.

---

About CCI, `tools.os_info` should never be used outside `build()` method which is likely the only method that is guaranteed to run on the build machine (right? I cannot think about any exception to this rule).